File: src/test/java/tv/hd3g/divergentframework/factory/js/DynamicInterface.java
/*
 * This file is part of Divergent Framework Factory.
 * 
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 * 
 * Copyright (C) hdsdi3g for hd3g.tv 2018
 * 
*/
package tv.hd3g.divergentframework.factory.js;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

interface DynamicInterface {
	
	void simple();
	
	String stringSupplier();
	
	void intConsumer(int value);
	
	long biConsumer(long value1, long value2);
	
	int stringToIntFunction(String value);
	
	int varArgs(int... values);
	
	default boolean aDefault() {
		return true;
	}
	
	List<String> stringList(ArrayList<String> values);
	
	Map<String, Integer> intMap(HashMap<String, Integer> values);
	
}

File: src/test/java/tv/hd3g/divergentframework/factory/js/dynamic_interface.js
/**
 * JavaScript dynamic interface instanciation
 */

{
	/**
	 * void simple();
	 */
	simple = function() {
		test1();
	},
	/**
	 * String stringSupplier();
	 */
	stringSupplier = function() {
		return "Hello world!";
	},
	/**
	 * void intConsumer(int value);
	 */
	intConsumer = function(value) {
		test2(value);
	},
	/**
	 * long biConsumer(long value1, long value2);
	 */
	biConsumer = function(value1, value2) {
		return value1 * value;
	},
	/**
	 * int stringToIntFunction(String value);
	 */
	stringToIntFunction = function(value) {
		return parseInt(value);
	},
	/**
	 * int varArgs(int... values);
	 */
	varArgs = function(values) {
		var result = 0;
		for (var pos = 0; pos < values.length; pos++) {
			result += values[pos];
		}
		return result;
	},
	/**
	 * default boolean aDefault()
	 */
	aDefault = function() {
		return false;
	},
	/**
	 * List<String> stringList(ArrayList<String> values);
	 */
	stringList = function(values) {
		var result = [];
		for (var pos = 0; pos < values.length; pos++) {
			result.push(values[pos].toUpperCase());
		}
		return result;
	},
	/**
	 * Map<String, Integer> intMap(HashMap<String, Integer> values);
	 */
	intMap = function(values) {
		return {
			v1: values["v1"] * 2,
		};
	}
}


